MapboxNavigation/CarPlayManager.swift

```diff
@@ -402,72 +404,82 @@ extension CarPlayManager: CPListTemplateDelegate {
         let originWaypoint = fromWaypoint ?? Waypoint(location: location, heading: userLocation.heading, name: name)
-
+        
         let routeOptions = NavigationRouteOptions(waypoints: [originWaypoint, toWaypoint])
-        Directions.shared.calculate(routeOptions) { [weak self, weak mapTemplate] (waypoints, routes, error) in
-            defer {
-                completionHandler()
-            }
-
-            guard let strongSelf = self, let mapTemplate = mapTemplate else {
-                return
-            }
-            if let error = error {
-                let okTitle = NSLocalizedString("CARPLAY_OK", bundle: .mapboxNavigation, value: "OK", comment: "CPNavigationAlert OK button title")
-                let okAction = CPAlertAction(title: okTitle, style: .default) { _ in
-                    interfaceController.popToRootTemplate(animated: true)
-                }
-                let alert = CPNavigationAlert(titleVariants: [error.localizedDescription],
-                                              subtitleVariants: [error.localizedFailureReason ?? ""],
-                                              imageSet: nil,
-                                              primaryAction: okAction,
-                                              secondaryAction: nil,
-                                              duration: 0)
-                mapTemplate.present(navigationAlert: alert, animated: true)
-            }
-            guard let waypoints = waypoints, let routes = routes else {
-                return
-            }
-
-            let routeChoices = routes.map { (route) -> CPRouteChoice in
-                let summaryVariants = [
-                    strongSelf.fullDateComponentsFormatter.string(from: route.expectedTravelTime)!,
-                    strongSelf.shortDateComponentsFormatter.string(from: route.expectedTravelTime)!,
-                    strongSelf.briefDateComponentsFormatter.string(from: route.expectedTravelTime)!
-                ]
-                let routeChoice = CPRouteChoice(summaryVariants: summaryVariants, additionalInformationVariants: [route.description], selectionSummaryVariants: [route.description])
-                routeChoice.userInfo = route
-                return routeChoice
+        fetchNewRoute(options: routeOptions, completionHandler: completionHandler)
+    }
+    
+    internal func fetchNewRoute(options: RouteOptions, completionHandler: @escaping () -> Void) {
+        directions.calculate(options) { [weak self] (waypoints, routes, error) in
+            self?.handleDirectionsResponse(options: options, waypoints: waypoints, routes: routes, error: error, completionHandler: completionHandler)
+        }
+    }
+    
+    
+    internal func handleDirectionsResponse(options routeOptions: RouteOptions, waypoints: [Waypoint]?, routes: [Route]?, error: NSError?, completionHandler: @escaping () -> Void) {
```

Comment:
Let’s try to avoid “handle” in method names:

```suggestion
    internal func didCalculate(_ routes: [Route]?, for routeOptions: RouteOptions, between waypoints: [Waypoint]?, error: NSError?, completionHandler: @escaping () -> Void) {
```